lib/modules/manager/bazel-module/fragments.ts

```diff
@@ -0,0 +1,327 @@
+import is from '@sindresorhus/is';
+import { z } from 'zod';
+import * as starlark from './starlark';
+
+// Fragment Schemas
+
+interface StringFragmentInterface {
```

Comment:
i don't like the interface suffix.

@zharinov can you help to simplify these schema stuff?

Reply:
@viceice I have an idea about how I can remove the `XXXInterface` declarations. Can you expand on what you don't like about the schema? Is it the duplication of the schema declarations in the `discriminatedUnion` and the separate `const` variables?

Reply:
My idea for removing the `XXXInterface` won't work. 

For context, I was asked to use a `discriminatedUnion` for the `ValueFragmentSchema`. Since it has a recursive data structure, one must define a type representing the output. Otherwise, you receive a type annotation error. 

```typescript
// ERROR: 7022: 'ValueFragmentSchema' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
export const ValueFragmentSchema = z.lazy(
  () =>
    z.discriminatedUnion('type', [
      z.object({ type: z.literal('string'), value: z.string() }),
      z.object({ type: z.literal('boolean'), value: z.boolean() }),
      z.object({
        type: z.literal('array'),
        items: ValueFragmentSchema.array(),
        isComplete: z.boolean(),
      }),
      z.object({
        type: z.literal('record'),
        children: z.record(z.string(), ValueFragmentSchema),
        isComplete: z.boolean(),
      }),
    ])
);
```

To define the type, I need to define an `interface` or `type` for the entire shape of the schema. Those are the `XXXInteferface` declarations.

I redefined the types defined in the `discriminatedUnion` in the `XXXFragmentSchema` constants to reference them in the `rules` module where I define rule-specific schemas. Ideally, I would define the individual schemas as constants and then reference them in the `discriminatedUnion`. Alas, the recursion does not allow this.

Unless someone has a better way to model this, I don't see a way to remove the `XXXInterface` declarations. However, I am not married to the name. I am happy to rename them. Would `XXXType` be more palatable?

Reply:
I think @zharinov is best to help here.

Reply:
I refactored `fragments.ts`, removing the classes, the recursive schemas, and the `XXXInterface` types.